### PR TITLE
Extension API, self-host codegen, generated extension declarations

### PR DESCRIPTION
**Pull request by andrewparmet**
## Summary

- Add proto2 extension API to the runtime with type-safe `message[ext]` access and generated `val Extendee.name` extension properties
- Fully self-host the codegen pipeline on protokt's own types
- Teach the codegen to emit Extension declarations from `extend` blocks

### Commit 1: Add extension API and promote unknownFields to Message interface
- `val unknownFields: UnknownFieldSet` on the `Message` interface
- Privatize `UnknownFieldSet` internal map behind `operator get`, `operator contains`, `forEach`
- `Extension<E, T>`, `RepeatedExtension<E, T>`, `ExtensionCodec` with all 17 wire type codecs
- Simplify `protokt-reflect`: delete `getUnknownFields()` kotlin-reflect hack
- API dump updates

### Commit 2: Add unit tests for extension API and interop demo
- `ExtensionTest`: all 17 codec types, encode round-trips, operator get, edge cases
- `ExtensionInteropTest`: reads protokt's own custom options from protobuf-java wire bytes and live descriptors

### Commit 3: Migrate codegen pipeline to protokt descriptor types
- New `protokt-bootstrap` module with checked-in generated code
- Replace all `DescriptorProtos.*` with `protokt.v1.google.protobuf.*` throughout the codegen
- Parse CodeGeneratorRequest/Response with protokt types
- Remove ExtensionRegistry, remove protobuf-java generation of protokt.proto
- GrpcKotlinGeneratorSupport: serialize at the grpc-kotlin boundary only

### Commit 4: Generate Extension declarations from extend blocks
- Codegen emits typed Kotlin extension properties for `extend` blocks
- Each extension produces `val Extendee.name: T?` backed by `Extension<E, T>`
- Regenerated bootstrap includes the 8 protokt option extensions
- Delete hand-written `ProtoktExtensions.kt`; call sites use generated properties

### Commit 5: Add regenerateBootstrap and verifyBootstrap Gradle tasks
- `regenerateBootstrap`: downloads well-known protos from protobuf GitHub, runs protoc + codegen
- `verifyBootstrap`: fails if checked-in files drift
- Wired into CI

### CODE REVIEW

File: protokt-codegen/build.gradle.kts [hunk 1]
```diff
@@ -103,15 +103,106 @@ tasks.withType<Test> {
...
+            val target = outputDir.get().file(proto).asFile
+            target.parentFile.mkdirs()
+            uri("$baseUrl/$proto").toURL().openStream().use { input ->
+                target.outputStream().use { output -> input.copyTo(output) }
+            }
+        }
+    }
+}
+
+val regenerateBootstrap by tasks.registering(Exec::class) {
```
Comment (ogolberg) [inline protokt-codegen/build.gradle.kts]: this (and the other two tasks) should probably be concrete task classes; they are too complex and have a bunch of doLast logic

File: protokt-codegen/build.gradle.kts [hunk 2]
```diff
@@ -103,15 +103,106 @@ tasks.withType<Test> {
...
+    val version = libs.versions.protobuf.java.get()
+    outputs.dir(outputDir)
+
+    doLast {
+        val baseUrl = "https://raw.githubusercontent.com/protocolbuffers/protobuf/v$version/src"
+        val protos = listOf(
+            "google/protobuf/descriptor.proto",
+            "google/protobuf/compiler/plugin.proto"
+        )
+        protos.forEach { proto ->
```
Comment (ogolberg) [inline protokt-codegen/build.gradle.kts]: this could be cacheable if defined as a proper task with version as an input

**Review by ogolberg — APPROVED** [review]
